cli/azd/cmd/package.go

```diff
@@ -0,0 +1,197 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/azure/azure-dev/cli/azd/cmd/actions"
+	"github.com/azure/azure-dev/cli/azd/internal"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/output/ux"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type packageFlags struct {
+	serviceName string
+	global      *internal.GlobalCommandOptions
+	*envFlag
+}
+
+func newPackageFlags(cmd *cobra.Command, global *internal.GlobalCommandOptions) *packageFlags {
+	flags := &packageFlags{
+		global:  global,
+		envFlag: newEnvFlag(cmd, global),
+	}
+
+	flags.Bind(cmd.Flags(), global)
+
+	return flags
+}
+
+func (pf *packageFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandOptions) {
+	local.StringVar(
+		&pf.serviceName,
+		"service",
+		"",
+		//nolint:lll
+		"Deploys a specific service (when the string is unspecified, all services that are listed in the "+azdcontext.ProjectFileName+" file are deployed).",
+	)
+	//deprecate:flag hide --service
+	_ = local.MarkHidden("service")
+}
+
+func newPackageCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "package <service>",
+		Short:  "Packages the application's code to be deployed to Azure.",
+		Hidden: true,
+	}
+	cmd.Args = cobra.MaximumNArgs(1)
+	return cmd
+}
+
+type packageAction struct {
+	flags          *packageFlags
+	args           []string
+	projectConfig  *project.ProjectConfig
+	projectManager project.ProjectManager
+	serviceManager project.ServiceManager
+	console        input.Console
+	formatter      output.Formatter
+	writer         io.Writer
+}
+
+func newPackageAction(
+	flags *packageFlags,
+	args []string,
+	projectConfig *project.ProjectConfig,
+	projectManager project.ProjectManager,
+	serviceManager project.ServiceManager,
+	console input.Console,
+	formatter output.Formatter,
+	writer io.Writer,
+) actions.Action {
+	return &packageAction{
+		flags:          flags,
+		args:           args,
+		projectConfig:  projectConfig,
+		projectManager: projectManager,
+		serviceManager: serviceManager,
+		console:        console,
+		formatter:      formatter,
+		writer:         writer,
+	}
+}
+
+type PackageResult struct {
+	Timestamp time.Time                                `json:"timestamp"`
+	Services  map[string]*project.ServicePackageResult `json:"services"`
+}
+
+func (pa *packageAction) Run(ctx context.Context) (*actions.ActionResult, error) {
```

Comment:
Can be a separate fast-follow, but we should have end-to-end tests for asserting the output for `Package` on each `language`

Reply:
The goal was to call the package command as a pre-req command for `up` and other actions.  If this is still the case then we need to expose this package action.